transaction-status/src/parse_token/extension/confidential_transfer.rs

```diff
@@ -103,14 +123,34 @@ pub(in crate::parse_token) fn parse_confidential_transfer_instruction(
             let proof_instruction_offset: i8 = empty_account_data.proof_instruction_offset;
             let mut value = json!({
                 "account": account_keys[account_indexes[0] as usize].to_string(),
-                "instructionsSysvar": account_keys[account_indexes[1] as usize].to_string(),
                 "proofInstructionOffset": proof_instruction_offset,
 
             });
             let map = value.as_object_mut().unwrap();
+            let offset = if proof_instruction_offset == 0 {
+                map.insert(
+                    "proofContextStateAccount".to_string(),
+                    json!(account_keys[account_indexes[1] as usize].to_string()),
+                );
+                2
+            } else {
+                map.insert(
+                    "instructionsSysvar".to_string(),
+                    json!(account_keys[account_indexes[1] as usize].to_string()),
+                );
+                if account_indexes.len() > 3 {
+                    map.insert(
+                        "recordAccount".to_string(),
+                        json!(account_keys[account_indexes[2] as usize].to_string()),
+                    );
+                    3
+                } else {
+                    2
+                }
+            };
             parse_signers(
```

Comment:
The function `parse_signers` uses unsafe indexing, but `check_num_token_accounts` does not guarantee safety if `offset` goes up to 3. Should we add an if statement to prevent panic? This seem to apply for all instructions with variable number of accounts...

Reply:
We should be safe since we check that there's at least one more account everywhere. Here, for example, the offset `3` is only returned if there's at least 4 accounts, so it's safe. And for the offset `2`, we've already checked that there's at least `3` accounts

Reply:
Is there a place where that isn't done?

Reply:
Oh yes you are right! It seems that in all the places, it is safe.